### PR TITLE
Support for Cluster in PusherOptions and small fixes

### PR DESCRIPTION
**Pull request by Misiu**
I've added support for Cluster inside PusherOptions. This allows to create correct host based on specified Cluster.
Rest of changes:
- I've updated dependencies:
  - Newtonsoft.Json to 9.0.1
  - WebSocket4Net to 0.14.1
- Removed unused `using` statements
- Fixed NuGet so now it restores packages when project is build

### CODE REVIEW

File: PusherClient/PusherOptions.cs [hunk 1]
```diff
@@ -4,6 +4,10 @@ public class PusherOptions
     {
         public bool Encrypted = false;
         public IAuthorizer Authorizer = null;
-        public string Host = "ws.pusherapp.com";
+        public string Cluster = "eu";
```
Comment (luismfonseca) [inline PusherClient/PusherOptions.cs]: You're changing the default cluster, it should be `mt1`.

Comment (Misiu) [inline PusherClient/PusherOptions.cs]: @luismfonseca I've just fixed that and updated my PR.

**Comment by Misiu** [comment]
Any news on this?

**Comment by hot-leaf-juice** [comment]
Hi @Misiu,

Sorry it's taken so long to get around to reviewing these changes. The actual code changes look great, but the configuration changes seem to be a regression. The project currently uses the more up to date [automatic package restore](http://docs.nuget.org/ndocs/consume-packages/package-restore) and you seem to have moved back to the old MSBuild-integrated restore. (Automatic restore doesn't use a .nuget folder, is simpler, and more robust. See [here](http://docs.nuget.org/ndocs/consume-packages/package-restore#migrating-to-automatic-restore) for details on migrating.)

If you could untrack the following, then we can get this merged. Otherwise I'll move your changes in to a new branch and get them merged without the nuget changes (with credit of course 🙂 ).

- everything in .nuget
- AuthHost/AuthHost.csproj
- ExampleApplication/ExampleApplication.csproj
- `<Import Project="$(SolutionDir)\.nuget\NuGet.targets"` etc in PusherClient/PusherClient.csproj
- pusher-dotnet-client.sln

Again thanks for this. No worries if you don't have time to make these changes, I'm happy to sort it out.

**Comment by Misiu** [comment]
@hot-leaf-juice thank You for reply. I'm not expert when it comes to nuget to Yes, if You'd like Your more than welcome to fix this.
I'll try to remember about that when making future PR's.

**Comment by hot-leaf-juice** [comment]
see #22

**Pull request closed without merging**